toffy/tiling_utils.py

```diff
@@ -928,12 +959,199 @@ def write_manual_to_auto_map(manual_to_auto_map, save_ann, mapping_path):
+    duplicate_auto_fovs = [
+        (af, tuple(mf_list)) for (af, mf_list) in auto_fov_mappings.items() if len(mf_list) > 1
+    ]
+
+    # sort auto FOVs alphabetically
+    duplicate_auto_fovs = sorted(duplicate_auto_fovs, key=lambda val: val[0])
+
+    return duplicate_auto_fovs
+
+
+def find_manual_auto_name_mismatches(manual_to_auto_map):
+    """Finds the manual FOVs with names that do not match their corresponding auto FOV
+
+    Args:
+        manual_to_auto_map (dict):
+            defines the mapping of manual to auto FOV names
+
+    Returns:
+        list:
+            contains tuples with elements:
+
+            - `str`: the manual FOV
+            - `str`: the corresponding auto FOV
+    """
+
+    # find the manual FOVs that don't match their corresponding closest_auto_fov name
+    # NOTE: this method maintains the original manual FOV ordering which is already sorted
+    manual_auto_mismatches = [
+        (k, v['closest_auto_fov'])
+        for (k, v) in manual_to_auto_map.items() if k != v['closest_auto_fov']
+    ]
+
+    return manual_auto_mismatches
+
+
+# TODO: potential type hinting candidate?
+def generate_validation_annot(manual_to_auto_map, check_dist=50,
+                              check_duplicates=True, check_mismatches=True):
+    """Finds problematic manual-auto FOV pairs and generates a warning message to display
+
+    The following potential sources of error can be requested by the user:
+
+    - Manual to auto FOV pairs that are of a distance greater than `check_dist` away
+    - Auto FOV names that have more than one manual FOV name mapping to it
+    - Manual to auto FOV pairs that don't have the same name
+
+    Args:
+        manual_to_auto_map (dict):
+            defines the mapping of manual to auto FOV names
+        check_dist (float):
+            if the distance between a manual-auto FOV pair exceeds this value, it will
+            be reported for a potential error, if `None` does not validate distance
+        check_duplicates (bool):
+            if `True`, report each auto FOV with 2 or more manual FOVs mapping to it
+        check_mismatches (bool):
+            if `True`, report manual FOVs that map to an auto FOV with a different name
+
+    Returns:
+        str:
+            describes the validation failures
+    """
+
+    # define the potential sources of error desired by user in the mapping
+    invalid_dist = find_manual_auto_invalid_dist(
+        manual_to_auto_map, check_dist
+    ) if check_dist is not None else []
+    duplicate_auto = find_duplicate_auto_mappings(
+        manual_to_auto_map
+    ) if check_duplicates else []
+    name_mismatch = find_manual_auto_name_mismatches(
+        manual_to_auto_map
+    ) if check_mismatches else []
+
+    # generate the annotation
+    warning_annot = ""
+
+    # add the manual-auto FOV pairs with distances greater than check_dist
+    # TODO: should we stick with pixels or should we convert to microns or stage coordinates?
```

Comment:
I think microns makes the most sense as the distance metric

Reply:
Ok, I'll rework the distance computations in `assign_closest_fovs` to account for this.